__tests__/app.test.js

```diff
@@ -221,4 +222,64 @@ describe("GET /api/articles/:article_id/comments", () => {
                 expect(response.body.comments.length).toEqual(0);
             });
     });
+});
+
+describe("POST /api/articles/:article_id/comments", () => {
+    test(`Return is an object, with keys: comment_id, votes, created_at, author, body`, () => {
+        return request(app)
+            .post("/api/articles/1/comments")
+            .send({ "username": "lurker", "body": "new sauce" })
+            .expect(200)
+            .then((response) => {
+                expect(response.body.comment).toEqual(expect.objectContaining(
+                    {
+                        comment_id: expect.any(Number),
+                        votes: expect.any(Number),
+                        created_at: expect.any(String),
+                        author: expect.any(String),
+                        body: expect.any(String),
+                        article_id: expect.any(Number)
+                    }));
+            });
+    });
+
+    test(`Bad request message sent when body property doesnt exist in request body.`, () => {
+        return request(app)
+            .post("/api/articles/1/comments")
+            .send({ "username": "lurker", "bodyyy": "new sauce" })
+            .expect(400)
+            .then((response) => {
+                expect(response.body.error).not.toBe(undefined);
+            });
+    });
+
+    test(`Bad request message sent when username property doesnt exist in request body.`, () => {
+        return request(app)
+            .post("/api/articles/1/comments")
+            .send({ "usernameee": "lurker", "body": "new sauce" })
+            .expect(400)
+            .then((response) => {
+                expect(response.body.error).not.toBe(undefined);
```

Comment:
Instead of checking that its not undefined I would check for the error message you're expecting?
Just incase you're getting a different error than the one you are expecting.